odoo/addons/base/models/ir_ui_view.py

```diff
@@ -568,6 +568,11 @@ def apply_inheritance_specs(self, source, specs_tree, inherit_id):
         # changes to apply to some parent architecture).
         specs = [specs_tree]
 
+        def _move_node(node):
+            to_move = self.locate_node(source, node)
+            remove_element(to_move)
+            return copy.deepcopy(to_move)
```

Comment:
The function name is not very appealing. It locates a node given a specification, remove it from the source, and return it. I suggest the following API:
```python
def extract(spec):
    ...
```
Please explain why you use `deepcopy` on the detached node. Can you not use the node after its removal?